WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -29,6 +29,9 @@ class CompetitionsController < ApplicationController
 
   def new
     @competition = Competition.new
+    if current_user.any_kind_of_delegate?
+      @competition.delegates |= [current_user]
```

Comment:
Actually although it saves the correct records to the database one of the test fails because of the duplicate record. I might leave it as |=

Here's the test that fails:

`      it 'clones a competition that they delegated' do
        # First, make ourselves the delegate of the competition we're going to clone.
        competition.delegates = [delegate]
        get :clone_competition, id: competition
        new_comp = assigns(:competition)
        expect(new_comp.id).to eq ""

        # Cloning a competition should clone its organizers.
        expect(new_comp.organizers.sort_by(&:id)).to eq []
        # When a delegate clones a competition, it should clone its organizers, and add
        # the delegate doing the cloning.
        expect(new_comp.delegates.sort_by(&:id)).to eq [delegate]
      end`

It's because the delegate is there twice.